grabbit/src/main/groovy/com/twcable/grabbit/client/batch/steps/workspace/DeleteBeforeWriteTasklet.groovy

```diff
@@ -0,0 +1,158 @@
+package com.twcable.grabbit.client.batch.steps.workspace
+
+import com.twcable.grabbit.jcr.JcrUtil
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.apache.sling.jcr.api.SlingRepository
+import org.springframework.batch.core.StepContribution
+import org.springframework.batch.core.scope.context.ChunkContext
+import org.springframework.batch.core.step.tasklet.Tasklet
+import org.springframework.batch.repeat.RepeatStatus
+
+import javax.jcr.Node
+import javax.jcr.Session
+
+/*
+ * Copyright 2015 Time Warner Cable, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@Slf4j
+@CompileStatic
+class DeleteBeforeWriteTasklet implements Tasklet {
+
+    private String jobPath
+
+    private Collection<String> excludePaths
+
+    private SlingRepository slingRepository
+
+    void setExcludePaths(String paths) {
+        final thisPathsString = paths?.trim()
+        if(!thisPathsString) {
+            excludePaths = []
+            return
+        }
+        Collection<String> theseExcludePaths = thisPathsString.split("\\*") as Collection
+        excludePaths = theseExcludePaths.collect { String excludePath ->
+            return (excludePath[0] != '/') ? "/${excludePath}".toString() : excludePath
+        }
```

Comment:
Even better, as I continue to think through this, would be
```groovy
def theExcludePaths = thisPathsString.split("\\*") as Collection<String>
def jobPathSlash = jobPath + '/'
if (theExcludePaths.any {!it.startsWith(jobPathSlash)}) throw new IllegalStateException("Not all exclude paths start with \"${jobPath}\": ${theExcludePaths}")
this.excludePaths = theExcludePaths.collect { it.substring(jobPathSlash.length()) }
```
then you can always treat the exclude paths as relative (since later they are converted that way anyway) **_and_** you've already validated that they are legit for the job path.